code/DDSCodeTester.cpp

```diff
@@ -4486,11 +4554,837 @@ void log_examples()
+        std::string in_value = "helloworld";
+        std::string out_value;
+        data->set_string_value(data->get_member_id_by_name("my_string"), in_value);
+        data->get_string_value(out_value, data->get_member_id_by_name("my_string"));
+        //!--
+    }
+    {
+        //!--CPP_ENUM
+        enum MyEnum : uint32_t
+        {
+            A,
+            B,
+            C
+        };
+
+        // Define a struct type to contain an enum
+        TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+        type_descriptor->kind(TK_STRUCTURE);
+        type_descriptor->name("EnumStruct");
+        DynamicTypeBuilder::_ref_type struct_builder {DynamicTypeBuilderFactory::get_instance()->
+                                                              create_type(type_descriptor)};
+        // Define the enum type
+        TypeDescriptor::_ref_type enum_type_descriptor {traits<TypeDescriptor>::make_shared()};
+        enum_type_descriptor->kind(TK_ENUM);
+        enum_type_descriptor->name("MyEnum");
+        DynamicTypeBuilder::_ref_type enum_builder {DynamicTypeBuilderFactory::get_instance()->
+                                                            create_type(enum_type_descriptor)};
+        // Add enum literals to the enum type
+        MemberDescriptor::_ref_type enum_member_descriptor {traits<MemberDescriptor>::make_shared()};
+        enum_member_descriptor->type(DynamicTypeBuilderFactory::get_instance()->get_primitive_type(TK_UINT32));
+        enum_member_descriptor->name("A");
+        enum_builder->add_member(enum_member_descriptor);
+        enum_member_descriptor = traits<MemberDescriptor>::make_shared();
+        enum_member_descriptor->type(DynamicTypeBuilderFactory::get_instance()->get_primitive_type(TK_UINT32));
+        enum_member_descriptor->name("B");
+        enum_builder->add_member(enum_member_descriptor);
+        enum_member_descriptor = traits<MemberDescriptor>::make_shared();
+        enum_member_descriptor->type(DynamicTypeBuilderFactory::get_instance()->get_primitive_type(TK_UINT32));
+        enum_member_descriptor->name("C");
+        enum_builder->add_member(enum_member_descriptor);
+        // Build the enum type
+        DynamicType::_ref_type enum_type = enum_builder->build();
+
+        // Add an enum member to the struct
+        MemberDescriptor::_ref_type member_descriptor {traits<MemberDescriptor>::make_shared()};
+        member_descriptor->name("my_enum");
+        member_descriptor->type(enum_type);
+        struct_builder->add_member(member_descriptor);
+        // Build the struct type
+        DynamicType::_ref_type struct_type {struct_builder->build()};
+        // Create dynamic data based on the struct type
+        DynamicData::_ref_type data {DynamicDataFactory::get_instance()->create_data(struct_type)};
+
+        // Set and retrieve values for an enum member
+        MyEnum in_value = MyEnum::C;
```

Comment:
I will not use a C++ enum. Why define a enumeration with DynamicTypes if then your are using a C++ enumation. Here I will set:
```suggestion
        uint32_t in_value {2}; // Selecting MyEnum::C which value is 2
```